src/enzyme_ad/jax/Passes/Passes.td

```diff
@@ -11,7 +11,7 @@
 
 include "mlir/Pass/PassBase.td"
 
-def CanonicalizeLoopsPass : Pass<"canonicalize-loops"> {
+def CanonicalizeLoopsPass : Pass<"canonicalize-loops", "::mlir::func::FuncOp"> {
```

Comment:
Do we need to enforce funcop or can we run more generally (perhaps inside the pass walk all function interfaces)

Reply:
yes we can use FuncOpInterface.